### PR TITLE
Fix date custom field formatting

### PR DESCRIPTION
**Pull request by vboctor**
Fixes #22926

### CODE REVIEW

**Comment by atrol** [comment]
This fixes #22926, but the fix which introduced the issue (#22428) introduces even more issues, e.g.#22937

Maybe it's better in short term (2.4.2, 2.5.0) to revert the changes from #22428, and to provide a proper fix in 2.6.0 with some more tests.

**Comment by vboctor** [comment]
@atrol I have fixed the email field formatting.  Will test all custom fields types tonight.

**Comment by vboctor** [comment]
@atrol This PR now includes fix for date, email, and single line string custom fields.

File: core/cfdefs/cfdef_standard.php [hunk 1]
```diff
@@ -261,12 +261,23 @@ function cfdef_prepare_list_value_for_email( $p_value ) {
 }
 
 /**
- * Format email address for email
- * @param string $p_value Value.
- * @return string
+ * Print the value of the email custom field.
+ * @param $p_value The database value
```
Comment (atrol) [inline core/cfdefs/cfdef_standard.php]: missing parameter type

File: core/cfdefs/cfdef_standard.php [hunk 2]
```diff
@@ -261,12 +261,23 @@ function cfdef_prepare_list_value_for_email( $p_value ) {
...
+ * @return string The html value.
```
Comment (atrol) [inline core/cfdefs/cfdef_standard.php]: functions does not return any value

File: plugins/MantisCoreFormatting/MantisCoreFormatting.php [hunk 3]
```diff
@@ -190,7 +190,7 @@ function formatted( $p_event, $p_string, $p_multiline = true ) {
 		$t_string = mention_format_text( $t_string, /* html */ true );
 
 		# Process Markdown
-		if( ON == $s_markdown && $p_multiline ) {
+		if( ON == $s_markdown ) {
```
Comment (atrol) [inline plugins/MantisCoreFormatting/MantisCoreFormatting.php]: This introduces side effects, e.g. check rendering of summary field on "My View" page. or "View Issue Details".
There are added new lines.

File: core/cfdefs/cfdef_standard.php (hunk 1 above)
Comment (vboctor) [inline core/cfdefs/cfdef_standard.php]: Fixed

File: core/cfdefs/cfdef_standard.php (hunk 2 above)
Comment (vboctor) [inline core/cfdefs/cfdef_standard.php]: Fixed

File: plugins/MantisCoreFormatting/MantisCoreFormatting.php (hunk 3 above)
Comment (vboctor) [inline plugins/MantisCoreFormatting/MantisCoreFormatting.php]: Fixed side effects

**Pull request merged**